src/main.cc

```diff
@@ -211,6 +212,8 @@ int main(int argc, char *argv[])
     QGCApplication* app = new QGCApplication(argc, argv, runUnitTests);
     Q_CHECK_PTR(app);
 
+    QApplication::setWindowIcon(QIcon(":/res/resources/icons/qgroundcontrol.ico"));
```

Comment:
Shouldn't it be fine everywhere? The other options for windows and osx set the icon for the binary which is also used for the app if this hasn't been set.